sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala

```diff
@@ -155,7 +159,12 @@ case class CustomShuffleReaderExec private(
 
   private lazy val cachedShuffleRDD: RDD[InternalRow] = {
     val executionId = sparkContext.getLocalProperty(SQLExecution.EXECUTION_ID_KEY)
-    SQLMetrics.postDriverMetricUpdates(sparkContext, executionId, metrics.values.toSeq)
+    SQLMetrics.postDriverMetricUpdates(sparkContext, executionId,
+      metrics.filter(_._1 != "partitionDataSize").values.toSeq)
+
+    if(!isLocalReader && shuffleStage.get.mapStats.isDefined) {
+      sendPartitionDataSizeMetrics(executionId, metrics.get("partitionDataSize").get)
```

Comment:
why not do `val partitionMetrics = metrics("partitionDataSize")` inside the method instead of passing as a parameter?